src/services/outliningElementsCollector.ts

```diff
@@ -83,16 +83,29 @@ namespace ts.OutliningElementsCollector {
         }
     }
 
+    function isRegionDelimiter(lineText: string) {
+        return lineText.match(/^\s*\/\/\s*#(end)?region(?:\s+(.*))?(?:\r)?$/);

```

Comment:
I would put the regexp in a constant to avoid parsing it again on every invocation.

Reply:
Fixed.